### ISSUE TITLE
Doesn't work with 2.1.0p10 and above

### ISSUE THREAD

**Issue by edu8** [MaximilianClemens/checkmk_fritzbox#7]
I recently upgraded my Check MK to 2.1.0p10 and got the message that there is missing monitoring data for the plugin.

**Comment by MaximilianClemens** [comment]
Hi @edu8,
Thanks for reporting that issue.
I didn't tested the check for 2.1.0 yet, but i'll try to do so this weekend and hopefully fix that issue.

Regards
Max

**Comment by MaximilianClemens** [comment]
Can you try [Release 0.5.7 ](https://github.com/MaximilianClemens/checkmk_fritzbox/releases/tag/v0.5.7).
I found a small Bug at the Rulepage, may this is the source of the Issue.

**Comment by edu8** [comment]
Hi @MaximilianClemens,
thanks for working on the issue directly. Unfortunately the new version is still not working. :(
Danke. ;-)

**Comment by interpreterle** [comment]
Hello @MaximilianClemens and @edu8,

I'm on CheckMK 2.1.0p13 with the check-mk-arm version. After changing the shebang in the python files from 
`#!/usr/bin/python3`
to
`#!/usr/bin/env python3`
the plugin works with my AVM FRITZ!DECT301 (FW: 05.02).

Greetings
Thomas

**Comment by ottl05** [comment]
> 

Hello @interpreterle ,
I´m trying to get it to run with 2.0.0p23.cre.
Could you tell me which files you changed?

thanks

**Comment by interpreterle** [comment]
Hello @ottl05,
after installing the files are located here:
local/share/check_mk/agents/special/agent_fritzbox_smarthome
local/share/check_mk/checks/agent_fritzbox_smarthome
local/share/check_mk/checks/fritzbox_smarthome
local/share/check_mk/web/plugins/metrics/metrics_check_fritzbox_smarthome.py
local/share/check_mk/web/plugins/wato/parameters_agent_fritzbox_smarthome.py

You need an omd restart after editing.

Greetings
Thomas

**Comment by MaximilianClemens** [comment]
Hi @interpreterle,

I changed the Shebang for all files as suggested by you, see version [v0.5.8](https://github.com/MaximilianClemens/checkmk_fritzbox/releases/tag/v0.5.8)
But I'm not sure, if that will fix the Problem @edu8 has. @ottl05 do you have the same problem?

What OS and checkmk do you use?
I'm using Ubuntu 20.04 with cre 2.1.0p11 and can not reproduce the issue :(

**Comment by edu8** [comment]
Hi @MaximilianClemens,
unfortunately the fix has not fixed my problem as you already expected. After installing the new release and restarting the site the monitoring data of the plugin is still missing.
I'm running Raspbian 11 Bullseye with currently CheckMK Raw 2.1.0p14, obviously an arm version as @interpreterle is using.
I hope that helps a bit to narrow it down more.
Edu8

**Comment by interpreterle** [comment]
Hello @MaximilianClemens.
I'm using RaspberryPi OS "Bullseye" (32 bit) with the Checkmk Raw Edition 2.1.0p13 from https://github.com/chrisss404/check-mk-arm
After re-investigating I guess the problem was the Shebang
`#!/usr/bin/python` (not: `#!/usr/bin/python3`) in some of your files because Debian Bullseye doesn't ship with `/usr/bin/python` by default anymore:
```
~$ su - omduser
~$ /usr/bin/python
-bash: /usr/bin/python: No such file or directory
~$ /usr/bin/python3
Python 3.9.2 (default, Mar 12 2021, 04:06:34)
[GCC 10.2.1 20210110] on linux
```

But CheckMK ships with it's own Python: 
```
~$ /usr/bin/env python3
Python 3.9.10 (main, Jul  8 2022, 11:02:38)
[GCC 10.2.1 20210110] on linux
```

The Ubuntu 20.04 Shebang may work because it's Debian Buster based:
```
/usr/bin/python
Python 2.7.18 (default, Jul  1 2022, 12:27:04)
[GCC 9.4.0] on linux2
```

Greetings
Thomas

**Comment by edu8** [comment]
Hi @interpreterle,
it seems we have about the same config (even our source for CheckMK is the same). Why on earth does the fix work for you but not for me? Have you installed anything python related extra on the system?
Anyway something changed since the fix is installed: CheckMK waited for the fritzbox plugin about a full second before and now the wait is about 0.1 seconds.
![image](https://user-images.githubusercontent.com/14105465/197044654-0314e4ef-a189-419a-9e12-25964ec1673f.png)

**Comment by interpreterle** [comment]
Hi @edu8, 
Maybe the difference is:
- I first upgraded my CheckMK instance to 2.1.x
- Found the plugin just a few days ago, so it was a fresh install on 2.1.x
- The plugin works not with my "normal" Fritzbox monitored host - I have created a new:
 Host properties: Checkmk agent / API integrations -> Configured API integrations, no Checkmk agent
 Search in setup for "Check state of Fritz!Box Smarthome Devices" and create rule with credentials and check Ignore SSL errors
 
I suggest to create a new host to check if the changes works - in my setup it does it but you have did an upgrade of CheckMK. 
If you still have problems please log in as your site user and do a:
` cmk -v yourhostname`

Greetings
Thomas

**Comment by MaximilianClemens** [comment]
@edu8 your error looks like this?
![image](https://user-images.githubusercontent.com/5416563/197062393-22dd2b9e-a9ac-4e2a-8967-1b9580ea457e.png)
Thanks @interpreterle, I think recreating the Rule "Check state of Fritz!Box Smarthome Devices" may fix the problem. I think the Rules got broken through the Update to 2.1, as the Plugin was using "RulespecGroupDatasourcePrograms" which were only working until cmk 2.0. Can you check if you can see your old rule? And if so, delete it and create a new one.

@interpreterle, what do you mean with "The plugin works not with my "normal" Fritzbox monitored host".
It should work with other agents on the same host:
![image](https://user-images.githubusercontent.com/5416563/197064548-76d28aa2-8db0-4b11-a555-95e5e57490bb.png)

**Comment by edu8** [comment]
Hi @MaximilianClemens,
that is the exact error i got. I deleted the old rule and created a new one. But still the error remains the same.
![image](https://user-images.githubusercontent.com/14105465/197123254-4756406e-0f85-41e5-b15d-5f97c01c0771.png)

**Comment by ottl05** [comment]
> Hi @interpreterle,
> 
> I changed the Shebang for all files as suggested by you, see version [v0.5.8](https://github.com/MaximilianClemens/checkmk_fritzbox/releases/tag/v0.5.8) But I'm not sure, if that will fix the Problem @edu8 has. @ottl05 do you have the same problem?
> 
> What OS and checkmk do you use? I'm using Ubuntu 20.04 with cre 2.1.0p11 and can not reproduce the issue :(

Hi,
I'm using RaspberryPi OS "Bullseye" (32 bit) with the Checkmk Raw Edition 2.1.0p14 from https://github.com/chrisss404/check-mk-arm.

---
I updated to 0.58 recreated thr rule and I´m receiving the values from the dect200 :-)

**Comment by edu8** [comment]
Hi,
after it seemed to work for @interpreterle and now also for @ottl05 with the same os and the same CheckMK fork from the same source I investigated what makes me different. So I went back to zero and started to check if everything is configured correct. 
Long story short: The rule in "Check state of Fritz!Box Smarthome Devices" was missing. Therefore CheckMK has no credentials and couldn't get any information. I have no clue why this rule went missing. It must have been there before otherwise it should never have worked and I can't imagine how I ever would have deleted it.
Anyways it's working for me as well. Thank you all for helping. :-)
